### DISCUSSION TITLE
Issue with Azerty Keyboard Configuration on Hyprland

### DISCUSSION THREAD

**Discussion by Raideeen** [hyprwm/Hyprland#3744] [Q&A] [answered]
#### Environment

- **Operating System**: Hyprland on Linux
- **Keyboard Layout**: Azerty

#### Description
I've recently transitioned from Windows to Linux and have encountered a couple of issues related to keyboard configuration on Hyprland, specifically with the Caps Lock behavior on an Azerty keyboard.

#### Issues

1. **Caps-Lock as Shift-Lock**:
   On Windows, the Caps Lock key functions as a Shift Lock, affecting all keys and not just letters. To replicate this behavior on Hyprland, I've set the following in `hyprland.conf`:
   ```conf
   kb_options = caps:shiftlock
   ```

2. **Caps-Lock Rapid Press Bug**:
   There's a longstanding Linux bug where rapid Caps Lock presses result in unintended multiple capital letters, like so: "THis is a SEntence". To counter this, I've tried editing the `capslock` file located in `/usr/share/X11/xkb/symbols` with the following changes:

   From this:
   ```xkb
   // This changes the <CAPS> key to become a Control modifier,
   // but it will still produce the Caps_Lock keysym.
   hidden partial modifier_keys
   xkb_symbols "ctrl_modifier" {
       replace key <CAPS> {
           type[Group1] = "ONE_LEVEL",
           symbols[Group1] = [ Caps_Lock ],
           actions[Group1] = [ SetMods(modifiers=Control) ]
       };
       modifier_map Control { <CAPS> };
   };
   ```

   To this:
   ```xkb
   // This changes the <CAPS> key to become a Control modifier,
   // but it will still produce the Caps_Lock keysym.
   hidden partial modifier_keys
   xkb_symbols "ctrl_modifier" {
       key <CAPS> {
           type="ALPHABETIC",
           repeat=No,
           symbols[Group1]= [ Caps_Lock, Caps_Lock ],
           actions[Group1]= [ LockMods(modifiers=Lock),
                              LockMods(modifiers=Shift+Lock,affect=unlock) ]
       };
   };
   ```

   However, this modification necessitates changing `caps:shiftlock` to `caps:ctrl_modifier` in `hyprland.conf`, which undoes my first fix.

#### Questions

- Is there a way to merge the behaviors of `caps:shiftlock` and `caps:ctrl_modifier` so that the Caps Lock can function as a Shift Lock without the bug of multiple capital letters?
- If a custom solution is required, could you provide guidance on creating a custom XKB symbol that achieves these objectives?

Any assistance or insight into resolving these issues would be greatly appreciated!

**Comment by sT4R3K** [comment, marked as answer]
Check this https://wiki.archlinux.org/title/Xorg/Keyboard_configuration#Switching_state_immediately_when_Caps_Lock_is_pressed
